pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go

```diff
@@ -0,0 +1,417 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodefeaturediscovery
+
+import (
+	"context"
+
+	secv1 "github.com/openshift/api/security/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	nfdconfig "github.com/kubernetes-sigs/node-feature-discovery-operator/pkg/config"
+)
+
+type controlFunc []func(n NFD) (ResourceStatus, error)
+
+type ResourceStatus int
+
+const (
+	Ready    ResourceStatus = 0
+	NotReady ResourceStatus = 1
+)
+
+func (s ResourceStatus) String() string {
+	names := [...]string{
+		"Ready",
+		"NotReady"}
+
+	if s < Ready || s > NotReady {
+		return "Unkown Resources Status"
+	}
+	return names[s]
+}
+
+func Namespace(n NFD) (ResourceStatus, error) {
+
+	state := n.idx
+	obj := n.resources[state].Namespace
+
+	found := &corev1.Namespace{}
+	logger := log.WithValues("Namespace", obj.Name, "Namespace", "Cluster")
+
+	logger.Info("Looking for")
+	err := n.rec.client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Not found, creating ")
+		err = n.rec.client.Create(context.TODO(), &obj)
+		if err != nil {
+			logger.Info("Couldn't create")
+			return NotReady, err
+		}
+		return Ready, nil
+	} else if err != nil {
+		return NotReady, err
+	}
+
+	logger.Info("Found, skpping update")
+
+	return Ready, nil
+}
```

Comment:
Are these functions used somewhere? I.e. `Namespace()` and all similar (`ServiceAccount()` et. al) below

Reply:
Woops, my bad 🙄These are obviously used in ´addResourcesControls()`